src/NativeScript/inlineFunctions.js

```diff
@@ -14,6 +14,103 @@ Object.assign(global, {
     },
+
+    // Decorators support    
+    __decorate(decorators, target, key, desc) {
+        var c = arguments.length;
+        var r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc;
+        var d;
+
+        if (typeof global.Reflect === "object" && typeof global.Reflect.decorate === "function") {
+            r = global.Reflect.decorate(decorators, target, key, desc);
+        }
+        else {
+            for (var i = decorators.length - 1; i >= 0; i--) {
+                if (d = decorators[i]) {
+                    r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
+                }
+            }
+        }
+        return c > 3 && r && Object.defineProperty(target, key, r), r;
+    },
+    __param(paramIndex, decorator) {
+        return function (target, key) {
+            decorator(target, key, paramIndex);
+        }
+    },  
+
+    // Decorators  
+    ObjC(name) {
+        var additionalParams = [];
+        var hasName = (name !== undefined && typeof name === "string");
+        var argsShift = hasName ? 1 : 0;
+        for (var i = argsShift; i < arguments.length; i++) {
+            additionalParams[i - argsShift] = arguments[i];
+        }
+
+        return function (target, propertyKey, descriptor) {
+            if (propertyKey === undefined) { // Class decorator
+                var implementation = {};
+                var baseTarget = Object.getPrototypeOf(target);
+                
+                for (var prop in target.prototype) {
+                    if (target.prototype.hasOwnProperty(prop)) {
+                        implementation[prop] = target.prototype[prop];
+                    }
+                }
+
+                var newName = hasName ? name : target.name;                
+                var extended = baseTarget.extend(implementation, {
+                    name: newName,
+                    protocols: additionalParams.length > 0 ? additionalParams : undefined,
+                    exposedMethods: target.ObjCExposedMethods
+                });
+                extended.name = newName;
+
+                // Add static members
+                for (var prop in target) {
+                    if (target.hasOwnProperty(prop) && prop !== "ObjCExposedMethods") {
+                        extended[prop] = target[prop];
+                    }
+                }
+
+                return extended;
+            }
+
+            // Method Decorator
+            if (!target.constructor.ObjCExposedMethods) {
+                target.constructor.ObjCExposedMethods = {};
+            }
+            if (!target.constructor.ObjCExposedMethods[propertyKey]) {
+                target.constructor.ObjCExposedMethods[propertyKey] = {};
+            }
+            target.constructor.ObjCExposedMethods[propertyKey].returns = additionalParams[0] || interop.types.void;
+
+            if (hasName && name !== propertyKey) {
```

Comment:
@PeterStaev As far as I understand, the idea behind this is to have ObjC friendly selector and JS-friendly method name simultaneously. Keep in mind that having a JS-unfriendly name is a concern only if we want to actually call the method from JS (often the exposed methods are only used in ObjC) and even without implicitly adding additional method you are still able to achieve the desired effect by adding it explicitly:
```
...
ObjCMethod(...)
public "selector:With:Parameters:" (x, y, z) {}

public selectorWithParameters() {
    return this["selector:With:Parameters:"].apply(this, arguments);
}
...
```
The explicit approach is more verbose (which is a con) but
1. It will eliminate the need of doing yet another special thing under the hood.
2. The method with the JS-friendly name may already exists in the prototype object.